### PR TITLE
[doc] Paimon metrics doc

### LINKED ISSUE #2001: [Feature] Document paimon metrics
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/incubator-paimon/issues) and found nothing similar.

### Motivation

Document paimon metrics in [PIP-3](https://cwiki.apache.org/confluence/display/PAIMON/PIP-3%3A+Introduce+metrics+for+Paimon)

### Solution

_No response_

### Anything else?

_No response_

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by schnappi17**
### Purpose

Linked issue: close #2001

### CODE REVIEW

File: docs/content/maintenance/metrics.md [hunk 1]
```diff
@@ -0,0 +1,387 @@
...
+
+In paimon's metrics system, metrics are updated and reported at different levels of granularity. Currently, the levels of **table** and **bucket** are provided, which means you can get metrics per table or bucket.
+
+There are three types of metrics provided in the paimon metric system, `Gauge`, `Counter`, `Histogram`.
+- `Gauge`: Provides a value of any type at a point in time.
+- `Counter`: Used to count values by incrementing and decrementing.
+- `Histogram`: Measure the statistical distribution of a set of values including the min, max, mean, standard deviation and percentile.
+
+Paimon has supported built-in metrics to measure operations of **commits**, **scans** and **compactions**, which can be bridged to any computing engine that supports, like flink, spark etc. Besides, when using flink to read and write, paimon has implemented some key standard flink connector metrics to measure the source latency and output of sink, see [FLIP-33: Standardize Connector Metrics](https://cwiki.apache.org/confluence/display/FLINK/FLIP-33%3A+Standardize+Connector+Metrics).
```
Comment (tsreaper) [inline docs/content/maintenance/metrics.md:38]: flink, spark, paimon should all be capitalized. flink -> Flink, spark -> Spark, paimon -> Paimon

File: docs/content/maintenance/metrics.md [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+
+# Metrics List
```
Comment (tsreaper) [inline docs/content/maintenance/metrics.md:40]:
```suggestion
## Metrics List
```

File: docs/content/maintenance/metrics.md [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+
+Below is lists of paimon built-in metrics. They are summarized into three types of metrics, scan metrics, commit metrics and compaction metrics. And also Flink source / sink metrics implemented are listed here.
+
+## Scan Metrics
```
Comment (tsreaper) [inline docs/content/maintenance/metrics.md:44]:
```suggestion
### Scan Metrics
```

File: docs/content/maintenance/metrics.md [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+        <tr>
+            <td>lastScanResultedTableFiles</td>
+            <td>Table</td>
+            <td>Gauge</td>
+            <td>Resulted table files in the last scan.</td>
+        </tr>
+    </tbody>
+</table>
+
+## Commit Metrics
```
Comment (tsreaper) [inline docs/content/maintenance/metrics.md:107]:
```suggestion
### Commit Metrics
```

File: docs/content/maintenance/metrics.md [hunk 5]
```diff
@@ -0,0 +1,387 @@
...
+        <tr>
+            <td>lastBucketsWritten</td>
+            <td>Table</td>
+            <td>Gauge</td>
+            <td>Number of buckets written in the last commit.</td>
+        </tr>
+    </tbody>
+</table>
+
+## Compaction Metrics
```
Comment (tsreaper) [inline docs/content/maintenance/metrics.md:218]:
```suggestion
### Compaction Metrics
```

File: docs/content/maintenance/metrics.md [hunk 6]
```diff
@@ -0,0 +1,387 @@
...
+        <tr>
+            <td>lastRewriteChangelogFileSize</td>
+            <td>Bucket</td>
+            <td>Gauge</td>
+            <td>Size of changelog files compacted in last compaction.</td>
+        </tr>
+    </tbody>
+</table>
+
+## Source Metrics (Flink)
```
Comment (tsreaper) [inline docs/content/maintenance/metrics.md:281]: Move into section `Bridging to Flink`.

File: docs/content/maintenance/metrics.md [hunk 7]
```diff
@@ -0,0 +1,387 @@
...
+        <tr>
+            <td>currentFetchEventTimeLag</td>
+            <td>Flink Source Operator</td>
+            <td>Gauge</td>
+            <td>Time difference between reading the data file and file creation.</td>
+        </tr>    
+    </tbody>
+</table>
+
+## Sink Metrics (Flink)
```
Comment (tsreaper) [inline docs/content/maintenance/metrics.md:302]: Same as above.

File: docs/content/maintenance/metrics.md [hunk 8]
```diff
@@ -0,0 +1,387 @@
...
+            <td>The output records per second.</td>
+        </tr>  
+    </tbody>
+</table>
+
+# Bridging To Flink
+
+Paimon has implemented bridging metrics to Flink's metrics system, which can be reported by Flink, and the lifecycle of metric groups are managed by Flink.
+
+Please join the `<scope>.<infix>.<metric_name>` to get the complete metric identifier when using Flink to access Paimon, `metric_name` can be got from [Metric List]({{< ref "maintenance/metrics#metrics-list" >}}).
```
Comment (tsreaper) [inline docs/content/maintenance/metrics.md:345]: Provide an example here. "For example, the identifier of metric `lastScanDuration` for table `MyTable` is xxxxxx".

File: docs/content/maintenance/metrics.md [hunk 9]
```diff
@@ -0,0 +1,399 @@
...
+        <tr>
+            <td>lastRewriteChangelogFileSize</td>
+            <td>Bucket</td>
+            <td>Gauge</td>
+            <td>Size of changelog files compacted in last compaction.</td>
+        </tr>
+    </tbody>
+</table>
+
+# Bridging To Flink
```
Comment (tsreaper) [inline docs/content/maintenance/metrics.md:281]:
```suggestion
## Bridging To Flink
```

Level 1 headers are only used for page titles.

File: docs/content/maintenance/metrics.md [hunk 10]
```diff
@@ -0,0 +1,399 @@
...
+        </tr>
+        <tr>
+            <td>Flink Sink Metrics</td>
+            <td>&lt;host&gt;.taskmanager.&lt;tm_id&gt;.&lt;job_name&gt;.&lt;committer_operator_name&gt;.&lt;subtask_index&gt;</td>
+            <td>-</td>
+        </tr>  
+    </tbody>
+</table>
+
+## Flink Connector Standard Metrics
```
Comment (tsreaper) [inline docs/content/maintenance/metrics.md:337]:
```suggestion
### Flink Connector Standard Metrics
```

File: docs/content/maintenance/metrics.md [hunk 11]
```diff
@@ -0,0 +1,399 @@
...
+
+When using Flink to read and write, Paimon has implemented some key standard Flink connector metrics to measure the source latency and output of sink, see [FLIP-33: Standardize Connector Metrics](https://cwiki.apache.org/confluence/display/FLINK/FLIP-33%3A+Standardize+Connector+Metrics). Flink source / sink metrics implemented are listed here.
+
+### Source Metrics (Flink)
```
Comment (tsreaper) [inline docs/content/maintenance/metrics.md:341]:
```suggestion
#### Source Metrics (Flink)
```

File: docs/content/maintenance/metrics.md [hunk 12]
```diff
@@ -0,0 +1,399 @@
...
+        <tr>
+            <td>currentFetchEventTimeLag</td>
+            <td>Flink Source Operator</td>
+            <td>Gauge</td>
+            <td>Time difference between reading the data file and file creation.</td>
+        </tr>    
+    </tbody>
+</table>
+
+### Sink Metrics (Flink)
```
Comment (tsreaper) [inline docs/content/maintenance/metrics.md:362]:
```suggestion
#### Sink Metrics (Flink)
```

File: docs/content/maintenance/metrics.md (hunk 9 above)
Comment (schnappi17) [reply docs/content/maintenance/metrics.md]: @tsreaper Got it, thanks for reminding~

**Review by tsreaper — APPROVED** [review]

**Pull request merged**